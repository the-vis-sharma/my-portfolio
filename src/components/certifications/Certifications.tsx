import React, { forwardRef } from "react";
import {
  Box,
  Button,
  Card,
  CardActions,
  CardContent,
  Divider,
  Grid,
  Link,
  Typography,
} from "@mui/material";
import { ArrowForward } from "@mui/icons-material";
import certifications from "./data";

const Certifications = forwardRef((_, ref) => {
  return (
    <Box
      ref={ref}
      sx={{ p: { xs: 2, sm: 4 }, pt: 8 }}
      id="certifications-section"
    >
      <Typography
        variant="h3"
        textAlign="center"
        gutterBottom
        color="primary"
        fontWeight="500"
        fontSize={{ xs: "2.125rem", sm: "3rem" }}
      >
        Certifications
      </Typography>
      <Grid container spacing={2} justifyContent="center" alignItems="center">
        {certifications.map((certification) => (
          <Grid
            key={certification.name}
            item
            md={6}
            sm={12}
            sx={{ position: "relative" }}
          >
            <Card data-testid="certification-card" variant="outlined">
              <CardContent>
                <Typography
                  gutterBottom
                  variant="h5"
                  component="div"
                  data-testid="certification-title"
                  color="primary"
                >
                  <Link
                    target="_blank"
                    href={certification.url}
                    style={{ textDecoration: "none" }}
                  >
                    {certification.name}
                  </Link>
                </Typography>
                <Typography
                  variant="body2"
                  color="text.secondary"
                  textAlign="justify"
                >
                  {certification.description}
                </Typography>
              </CardContent>
              <Divider />
              <CardActions
                sx={{ display: "flex", justifyContent: "space-between" }}
              >
                <Button
                  variant="text"
                  disabled
                  sx={{ textTransform: "capitalize" }}
                >
                  {`${certification.vendor} | ${certification.duration}`}
                </Button>
                <Button
                  target="_blank"
                  href={certification.url}
                  endIcon={<ArrowForward />}
                  variant="text"
                >
                  View
                </Button>
              </CardActions>
            </Card>
          </Grid>
        ))}
      </Grid>
    </Box>
  );
});

export default Certifications;
