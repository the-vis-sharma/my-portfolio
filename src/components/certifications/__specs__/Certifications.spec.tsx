import { render } from "@testing-library/react";
import Certifications from "../Certifications";

it("should verify snapshot", () => {
  const { asFragment } = render(<Certifications />);
  expect(asFragment()).toMatchSnapshot();
});
