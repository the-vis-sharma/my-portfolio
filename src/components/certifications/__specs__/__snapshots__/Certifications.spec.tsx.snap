// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should verify snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-eeo3ga"
    id="certifications-section"
  >
    <h3
      class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom css-1oe6gg4-MuiTypography-root"
    >
      Certifications
    </h3>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-6s1xft-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-6 css-kgax03-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1v5z4dq-MuiPaper-root-MuiCard-root"
          data-testid="certification-card"
        >
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1y15g74-MuiTypography-root"
              data-testid="certification-title"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
                href="https://www.credential.net/c9bf219d-3d0f-4072-94e4-894db446b28b"
                rel="noreferrer"
                style="text-decoration: none;"
                target="_blank"
              >
                Google Cloud Certified Professional Cloud Architect
              </a>
            </div>
            <p
              class="MuiTypography-root MuiTypography-body2 css-b3lvtk-MuiTypography-root"
            >
              A Google Cloud Certified - Professional Cloud Architect enables organizations to leverage Google Cloud technologies. Through an understanding of cloud architecture and Google technology, this individual can design, develop, and manage robust, secure, scalable, highly available, and dynamic solutions to drive business objectives.
            </p>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
          />
          <div
            class="MuiCardActions-root MuiCardActions-spacing css-1vbr7to-MuiCardActions-root"
          >
            <button
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root Mui-disabled  css-1poohah-MuiButtonBase-root-MuiButton-root"
              disabled=""
              tabindex="-1"
              type="button"
            >
              Google | June 2021 - June 2023
            </button>
            <a
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              href="https://www.credential.net/c9bf219d-3d0f-4072-94e4-894db446b28b"
              rel="noreferrer"
              tabindex="0"
              target="_blank"
            >
              View
              <span
                class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ArrowForwardIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-6 css-kgax03-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1v5z4dq-MuiPaper-root-MuiCard-root"
          data-testid="certification-card"
        >
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1y15g74-MuiTypography-root"
              data-testid="certification-title"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
                href="https://www.youracclaim.com/badges/acc77122-d057-4eac-a133-1a1b4e86edc3?source=linked_in_profile"
                rel="noreferrer"
                style="text-decoration: none;"
                target="_blank"
              >
                AWS Certified Cloud Practitioner
              </a>
            </div>
            <p
              class="MuiTypography-root MuiTypography-body2 css-b3lvtk-MuiTypography-root"
            >
              Earners of this certification have a fundamental understanding of IT services and their uses in the AWS Cloud. They demonstrated cloud fluency and foundational AWS knowledge. Badge owners are able to identify essential AWS services necessary to set up AWS-focused projects.
            </p>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
          />
          <div
            class="MuiCardActions-root MuiCardActions-spacing css-1vbr7to-MuiCardActions-root"
          >
            <button
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root Mui-disabled  css-1poohah-MuiButtonBase-root-MuiButton-root"
              disabled=""
              tabindex="-1"
              type="button"
            >
              AWS | August 2020 - August 2023
            </button>
            <a
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              href="https://www.youracclaim.com/badges/acc77122-d057-4eac-a133-1a1b4e86edc3?source=linked_in_profile"
              rel="noreferrer"
              tabindex="0"
              target="_blank"
            >
              View
              <span
                class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ArrowForwardIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-6 css-kgax03-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1v5z4dq-MuiPaper-root-MuiCard-root"
          data-testid="certification-card"
        >
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1y15g74-MuiTypography-root"
              data-testid="certification-title"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
                href="https://www.youracclaim.com/badges/497502bd-2058-45a5-9dad-3abe690c8d57?source=linked_in_profile"
                rel="noreferrer"
                style="text-decoration: none;"
                target="_blank"
              >
                Microsoft Certified: Azure Data Scientist Associate
              </a>
            </div>
            <p
              class="MuiTypography-root MuiTypography-body2 css-b3lvtk-MuiTypography-root"
            >
              Earning the Azure Data Scientist Associate certification validates the skills and knowledge to apply Azureâ€™s machine learning techniques to train, evaluate, and deploy models that solve business problems.
            </p>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
          />
          <div
            class="MuiCardActions-root MuiCardActions-spacing css-1vbr7to-MuiCardActions-root"
          >
            <button
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root Mui-disabled  css-1poohah-MuiButtonBase-root-MuiButton-root"
              disabled=""
              tabindex="-1"
              type="button"
            >
              Microsoft | January 2021 - January 2023
            </button>
            <a
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              href="https://www.youracclaim.com/badges/497502bd-2058-45a5-9dad-3abe690c8d57?source=linked_in_profile"
              rel="noreferrer"
              tabindex="0"
              target="_blank"
            >
              View
              <span
                class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ArrowForwardIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-6 css-kgax03-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1v5z4dq-MuiPaper-root-MuiCard-root"
          data-testid="certification-card"
        >
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1y15g74-MuiTypography-root"
              data-testid="certification-title"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
                href="https://www.youracclaim.com/badges/8d0994cf-2fd0-4092-81f9-416cd0985fe0/public_url"
                rel="noreferrer"
                style="text-decoration: none;"
                target="_blank"
              >
                Microsoft Certified: Azure Fundamentals
              </a>
            </div>
            <p
              class="MuiTypography-root MuiTypography-body2 css-b3lvtk-MuiTypography-root"
            >
              Earners of the Azure Fundamentals certification have demonstrated foundational level knowledge of cloud services and how those services are provided with Microsoft Azure.
            </p>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
          />
          <div
            class="MuiCardActions-root MuiCardActions-spacing css-1vbr7to-MuiCardActions-root"
          >
            <button
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root Mui-disabled  css-1poohah-MuiButtonBase-root-MuiButton-root"
              disabled=""
              tabindex="-1"
              type="button"
            >
              Microsoft | March 2021
            </button>
            <a
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              href="https://www.youracclaim.com/badges/8d0994cf-2fd0-4092-81f9-416cd0985fe0/public_url"
              rel="noreferrer"
              tabindex="0"
              target="_blank"
            >
              View
              <span
                class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ArrowForwardIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-6 css-kgax03-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1v5z4dq-MuiPaper-root-MuiCard-root"
          data-testid="certification-card"
        >
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1y15g74-MuiTypography-root"
              data-testid="certification-title"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
                href="https://confirm.udacity.com/PRCEZS2R"
                rel="noreferrer"
                style="text-decoration: none;"
                target="_blank"
              >
                IntelÂ® Edge AI For IoT Developers Nanodegree
              </a>
            </div>
            <p
              class="MuiTypography-root MuiTypography-body2 css-b3lvtk-MuiTypography-root"
            >
              Earner of this certificate knows the IntelÂ® Distribution of OpenVINOâ„¢ Toolkit to fast-track development of computer vision and deep learning inference applications and utilize the IntelÂ® DevCloud for the Edge to test model performance on the various hardware types.
            </p>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
          />
          <div
            class="MuiCardActions-root MuiCardActions-spacing css-1vbr7to-MuiCardActions-root"
          >
            <button
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root Mui-disabled  css-1poohah-MuiButtonBase-root-MuiButton-root"
              disabled=""
              tabindex="-1"
              type="button"
            >
              Udacity | July 2020
            </button>
            <a
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              href="https://confirm.udacity.com/PRCEZS2R"
              rel="noreferrer"
              tabindex="0"
              target="_blank"
            >
              View
              <span
                class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ArrowForwardIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
